src/box/lua/config/applier/sharding.lua

```diff
@@ -0,0 +1,71 @@
+local log = require('internal.config.utils.log')
+local fiber = require('fiber')
+_G.vshard = nil
+
+local function vshard_cfg(configdata)
+    local roles = configdata:get('sharding.roles')
+    if roles == nil then
+        return
+    end
+    if _G.vshard == nil then
+        _G.vshard = require('vshard')
+    end
+    for _, role in pairs(roles) do
+        local cfg = configdata:sharding()
+        cfg.listen = box.cfg.listen
+        cfg.read_only = box.cfg.read_only
+        cfg.replication = box.cfg.replication
```

Comment:
I think that this workaround should be commented, ideally with a link to vshard's issue.

Reply:
Added a comment.